src/main/java/teammates/ui/controller/AdminSearchPageAction.java

```diff
@@ -266,17 +266,15 @@ private String findAvailableInstructorGoogleIdForCourse(String courseId){
         if(instructorList == null || instructorList.isEmpty()){
             return googleId;
         }
-        
-        for(InstructorAttributes instructor : instructorList){
-          
-            if(instructor.googleId != null){
-               googleId = instructor.googleId;
-               if(instructor.isAllowedForPrivilege(Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER)){            
-                   break;
-               } 
-            }            
+
+        for (InstructorAttributes instructor : instructorList) {
+
+            if (instructor.isRegistered() && instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR)) {
```

Comment:
Can we look for the instructor with co-owner, then if we can't find one, then fall back to one  with the INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR permission? Perhaps this can be done as a separate lower priority issue in future.

While I generally agree with your logic, the admin shouldn't need to modify the instructor's permissions while using masquarade mode to look around in the course, so ideally, the instructor with highest permissions should be used here.